### PR TITLE
202306-notebook Ingestion of flight data as events [DNM]

### PR DESCRIPTION
**Pull request by petermarshallio**
New notebook that uses the sample flight data to walk through modeling principles while turning it into Event data rather than leaving it as Session data.

<hr>

This PR has:

- [X] been self-reviewed.
- [X] been tested in a test Druid cluster.
- [ ] been tested on the Notebook Docker

### CODE REVIEW

**Comment by petermarshallio** [comment]
Another notebook for y'all to try out!

**Review by techdocsmith — COMMENTED** [review]
some first impressions

File: examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb [hunk 1]
```diff
@@ -0,0 +1,807 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "e79d7d48-b403-4b9e-8cc6-0f0accecac1f",
+   "metadata": {},
+   "source": [
+    "# Data modeling and ingestion principles - creating Events from Druid's sample flight data\n",
+    "\n",
+    "Druid's data loader allows you to quickly ingest sample carrier data into a `TABLE`, giving you an easy way to learn about the SQL functions that are available. It's also a great place to start understanding how data modeling for event analytics in a real-time database differs from modeling you'd apply in other databases, as well as being small enough to safely see - and try out - different data layout designs safely.\n",
```
[the remark this answers is not in the document]
Comment (petermarshallio) [reply examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb]: Intro re-write done to swing this back towards what the notebook is about!

File: examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb [hunk 2]
```diff
@@ -0,0 +1,807 @@
...
+    "\n",
+    "In this notebook, you'll walk through creating a table of events out of the sample data set, applying data modeling principles as you go. At the end you'll have a `TABLE` called \"flight-events\" that you can then use as you continue your learning in Apache Druid.\n",
+    "\n",
+    "## Prerequisites\n",
```
[the remark this answers is not in the document]
Comment (petermarshallio) [reply examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb:14]: Prereqs updated thanks to @writer-jill

File: examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb [hunk 3]
```diff
@@ -0,0 +1,807 @@
...
+    "\n",
+    "In order to use this notebook, you'll need access to a small Druid deployment.\n",
```
[the remark this answers is not in the document]
Comment (petermarshallio) [reply examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb]: As above - pre-reqs standardized.

File: examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb [hunk 4]
```diff
@@ -0,0 +1,807 @@
...
+    "\n",
+    "It's a good idea to test ingesting the data \"as is\" on that cluster to make sure it's operational before you get going.\n",
```
[the remark this answers is not in the document]
Comment (petermarshallio) [reply examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb]: Updated.

File: examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb [hunk 5]
```diff
@@ -0,0 +1,807 @@
...
+    "\n",
+    "## Getting started\n",
```
[the remark this answers is not in the document]
Comment (petermarshallio) [reply examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb:28]: Reworded.

File: examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb [hunk 6]
```diff
@@ -0,0 +1,807 @@
...
+    "display = druid.display\n",
+    "sql_client = druid.sql"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "596eed1f-c47f-48cc-a537-5703c7eefc38",
+   "metadata": {},
+   "source": [
+    "## Apply modeling principles\n",
```
[the remark this answers is not in the document]
Comment (petermarshallio) [reply examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb:64]: Added text after each heading.

File: examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb [hunk 7]
```diff
@@ -0,0 +1,807 @@
...
+    "\n",
+    "### Principle 1 - create the right `TABLE` for the right query\n",
+    "\n",
+    "#### Finding the Events\n",
+    "\n",
+    "Let's take a look at the data we have. Using the Druid Console you can preview the data you want to load.\n",
+    "\n",
+    "1. Open the console\n",
```
[the remark this answers is not in the document]
Comment (petermarshallio) [reply examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb:72]: Shifted this into an admonition, and instead refer straight to the dataset definition. As is, the ingestion takes too long to run (attention killer) for me to pull this into the Druid service in order to show a sample of the rows.

File: examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb [hunk 8]
```diff
@@ -0,0 +1,807 @@
...
+    "* Data about the departure\n",
+    "* Data about takeoff (wheels off)\n",
+    "* Data about landing (wheels on)\n",
+    "* Data about arrival\n",
+    "\n",
+    "There is also Taxi In and Taxi Out data, and also information about cancellation. An opportunity here for Event analytics is to ask for more data about these events, either as part of this row or as individual events themselves (making our job easier!).\n",
+    "\n",
+    "Far to the right is data about diversions, but no easy way to calculate _when_ this happened so that we can create events.\n",
+    "\n",
+    "There are also a set of dimensions that have be calculated ahead of time for us. This is possibly because the databases used on top of this data can't calculate this on-the-fly in the way that Druid can; as with any data set, we need to decide whether to keep or remove them.\n",
```
[the remark this answers is not in the document]
Comment (petermarshallio) [reply examples/quickstart/jupyter-notebooks/notebooks/02-ingestion/XX-example-flightdata-events.ipynb:92]: Removed this line.

**Review by sergioferragut — CHANGES_REQUESTED** [review]
The declaration of "sql_client" at the beginning is missing, so none of the cells that use it are working.

**Comment by petermarshallio** [comment]
Thanks for that catch @sergioferragut !! Fixed and made some other little updates.

**Pull request closed without merging**